clients/python/src/model_registry/_client.py

```diff
@@ -75,16 +75,21 @@ def register_model(
         model_format_name: str,
         model_format_version: str,
         version: str,
-        author: str | None = None,
-        description: str | None = None,
         storage_key: str | None = None,
         storage_path: str | None = None,
         service_account_name: str | None = None,
+        author: str | None = None,
+        description: str | None = None,
         metadata: dict[str, ScalarType] | None = None,
     ) -> RegisteredModel:
         """Register a model.
 
-        Either `storage_key` and `storage_path`, or `service_account_name` must be provided.
+        This registers a model in the model registry. The model is not downloaded, and has to be stored prior to
+        registration.
+
+        Most models can be registered using their URI, along with optional connection-specific parameters, `storage_key`
+        and `storage_path` or, simply a `service_account_name`.
+        However it's advised to use the `utils.s3_uri_from` helper when using S3 object storage data connections.
```

Comment:
```suggestion
        URI builder utilities are recommended when referring to specialized storage; for example `utils.s3_uri_from` helper when using S3 object storage data connections.
```

what about something along this lines?